### PR TITLE
🧪 Add unit tests for SkillInstall

### PR DESCRIPTION
**Pull request by arran4**
🎯 **What:** The testing gap addressed was the missing tests for `SkillInstall` function in `skills/skill_install.go`. Added 7 new test cases to cover all identified edge cases.
📊 **Coverage:** The scenarios now tested include: empty source, empty name (name inference), fetch failure, missing SKILL.md, already installed skill, directory creation (mkdir) failure, and metadata write failure.
✨ **Result:** Test coverage for the `SkillInstall` function increased to 90%, and overall coverage for the `skills` package increased to 55.9%, improving test coverage and reliability for skill installation.

---
*PR created automatically by Jules for task [17100335828489872394](https://jules.google.com/task/17100335828489872394) started by @arran4*

### CODE REVIEW

File: skills/skill_test.go [hunk 1]
```diff
@@ -51,6 +51,124 @@ func TestSkillInstall(t *testing.T) {
...
+func TestSkillInstall_EmptySource(t *testing.T) {
+	err := SkillInstall("", "test_skill", "project", "")
+	if err == nil || err.Error() != "source is required" {
+		t.Fatalf("Expected error 'source is required', got: %v", err)
+	}
+}
+
+func TestSkillInstall_EmptyName(t *testing.T) {
+	tempSource := t.TempDir()
+	skillMd := filepath.Join(tempSource, "SKILL.md")
```
Comment (arran4) [inline skills/skill_test.go:62]: @jules * https://arran4.github.io/blog/post/2026/008-memory-fs-for-testing/
* https://arran4.github.io/blog/post/2026/020-optional-dependency-injection-via-type-switched-variadic-args/
* https://arran4.github.io/blog/post/2026/018-txtar-patterns-for-agents-v2/

Comment (arran4) [reply skills/skill_test.go:62]: @jules no fs in test at a minimum
